engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/analyse/alias/Graph.scala

```diff
@@ -675,6 +675,17 @@ object Graph {
 
       isDirectChildOf || isChildOfChildren
     }
+
+    private def removeScope(scope: Scope): Unit = {
+      childScopes = childScopes.filter(_ != scope)
```

Comment:
Just a _threading note_: this is probably not thread safe... in case we want to evaluate _visualizations_ in parallel, we may face issues like this at various places.

Reply:
Yeah, we are likely to open a can of worms.